src/pages/App.tsx

```diff
@@ -80,16 +82,60 @@ function getCurrentPageFromLocation(locationPathname: string): PageName | undefi
   }
 }
 
+// Get browser being used, code comes from: https://developer.mozilla.org/en-US/docs/Web/API/Window/navigator.
+function getBrowser(): string {
+  const sUsrAg = navigator.userAgent
+  // The order matters here, and this may report false positives for unlisted browsers.
+  if (sUsrAg.indexOf('Firefox') > -1) {
+    return BROWSER.FIREFOX
+    // "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:61.0) Gecko/20100101 Firefox/61.0"
+  } else if (sUsrAg.indexOf('SamsungBrowser') > -1) {
+    return BROWSER.SAMSUNG
+    // "Mozilla/5.0 (Linux; Android 9; SAMSUNG SM-G955F Build/PPR1.180610.011) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/9.4 Chrome/67.0.3396.87 Mobile Safari/537.36
+  } else if (sUsrAg.indexOf('Opera') > -1 || sUsrAg.indexOf('OPR') > -1) {
+    return BROWSER.OPERA
+    // "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.102 Safari/537.36 OPR/57.0.3098.106"
+  } else if (sUsrAg.indexOf('Trident') > -1) {
+    return BROWSER.INTERNET_EXPLORER
+    // "Mozilla/5.0 (Windows NT 10.0; WOW64; Trident/7.0; .NET4.0C; .NET4.0E; Zoom 3.6.0; wbx 1.0.0; rv:11.0) like Gecko"
+  } else if (sUsrAg.indexOf('Edge') > -1) {
+    return BROWSER.EDGE
+    // "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.36 Edge/16.16299"
+  } else if (sUsrAg.indexOf('Edg') > -1) {
+    return BROWSER.EDGE_CHROMIUM
+    // Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36 Edg/91.0.864.64
+  } else if (sUsrAg.indexOf('Chrome') > -1) {
+    return BROWSER.CHROME
+    // "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/66.0.3359.181 Chrome/66.0.3359.181 Safari/537.36"
+  } else if (sUsrAg.indexOf('Safari') > -1) {
+    return BROWSER.SAFARI
+    // "Mozilla/5.0 (iPhone; CPU iPhone OS 11_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/11.0 Mobile/15E148 Safari/604.1 980x1306"
+  } else {
+    return BROWSER.UNKNOWN
+  }
+}
+
 export default function App() {
   const { pathname } = useLocation()
   const currentPage = getCurrentPageFromLocation(pathname)
+  const isDarkMode = useIsDarkMode()
+
   useAnalyticsReporter()
   initializeAnalytics()
 
   useEffect(() => {
     window.scrollTo(0, 0)
   }, [pathname])
 
+  useEffect(() => {
+    // TODO(zzmp): add web vitals event properties to app loaded event.
+    sendAnalyticsEvent(EventName.APP_LOADED)
+    user.set(CUSTOM_USER_PROPERTIES.BROWSER, getBrowser())
+    user.set(CUSTOM_USER_PROPERTIES.DARK_MODE, isDarkMode)
+    user.set(CUSTOM_USER_PROPERTIES.SCREEN_RESOLUTION_HEIGHT, window.screen.height)
+    user.set(CUSTOM_USER_PROPERTIES.SCREEN_RESOLUTION_WIDTH, window.screen.width)
+  }, [isDarkMode])
```

Comment:
you want a separate hook just for `APP_LOADED`, otherwise it will resent whenever darkMode is changed. i guess i would leave all the screen resolution stuff in this hook for APP_LOADED, then make a new one for dark mode.

Reply:
ah good point / catch, thanks. done.